media-proxy/src/proxy_context.cc

```diff
@@ -25,49 +19,45 @@ ProxyContext::ProxyContext(void)
     mTcpCtrlPort = 8002;
 }
 
-ProxyContext::ProxyContext(std::string rpc_addr, std::string tcp_addr)
-try : mRpcCtrlAddr(rpc_addr), mTcpCtrlAddr(tcp_addr), mSessionCount(0) {
-    vector<string> sub_str;
-    istringstream f(mTcpCtrlAddr);
-    string s;
-
-    while (getline(f, s, ':')) {
-        sub_str.push_back(s);
-    }
-
-    if (sub_str.size() != 2) {
-        INFO("Illegal TCP listen address.");
+ProxyContext::ProxyContext(std::string_view rpc_addr, std::string_view tcp_addr)
```

Comment:
Have you changed the class from `std::string` to `std::string_view` only to be able to use it's methods in this constructor? I suggest not to change the classes everywhere but to use this class locally in the constructor only.

Reply:
No. I have changed it to `std::string_view` as this is the proper and fastest wat of passing `std::string` since `C++17`.
I do not even mention that the constructors like for ex. copy constructor should be explicitly disabled and it is not. Here in the overloaded version of constructor the main thing that took my attention was using try(), catch() statement inside it. 
Such construct - even if made possible by the standard itself - should no be used when constructing main class of the whole application. I do not even mention how the reverse stack unwrapping in case of unhandled exception would go.
So to sum this part up, I removed the `try` and `catch` as main constructor (initialization) should be allowed to fail and propagate to the caller. I also changed `string` to `string_view` - the first one in this form always does hard copy of full string when passed and the second one is secure no-copy equivalent (mostly).

Reply:
Also here is a simple speed comparison for both of them: 
![image](https://github.com/OpenVisualCloud/Media-Communications-Mesh/assets/89970744/c0cb662d-74e4-4464-addb-2f807fcf3b78)
[https://stackoverflow.com/a/40129046]

Reply:
Agree. Thanks for explaining that.